spec/requests/users_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe "Users", type: :request do
+
+  describe "User Registration" do
+    it "allows a user to sign up" do
+      post user_registration_path, params: {
+        user: {
+          user_name: "test",
+          phone: "1234567",
+          email: "testing@sample.com",
+          password: "password",
+          password_confirmation: "password"
+        }
+      }
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe "User Sign In" do
+    before do
+      @user = FactoryBot.create(:user) 
+    end
+
+    it "allows a user to sign in" do
+      post user_session_path, params: {
+        user: {
+          email: @user.email,
+          password: @user.password
+        }
+      }
+      expect(response).to redirect_to(root_path)
```

Comment:
Same thing here, you'd want to be sure that the user was actually signed in, not just sent to the root path. This can be done with something like checking that a successful flash message was returned, the correct status code was given back, etc.